### PR TITLE
Call transaction handlers after the prepare phase, not commit phase

### PR DESCRIPTION
**Pull request by gimantha**
Fixes ballerina-platform/ballerina-lang#29192

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -2,7 +2,7 @@ org.gradle.caching=true
 puppycrawlCheckstyleVersion=8.18
 group=org.ballerinalang
 version=1.0.10-SNAPSHOT
-ballerinaLangVersion=2.0.0-alpha7-SNAPSHOT
+ballerinaLangVersion=2.0.0-alpha7-20210326-211300-6ccfb47f
```
Comment (grainier) [inline gradle.properties:5]: This needs to be updated, to include lang changes of https://github.com/ballerina-platform/ballerina-lang/pull/29659

Comment (grainier) [inline gradle.properties:5]:
```suggestion
ballerinaLangVersion=2.0.0-alpha7-20210327-165900-8a5ba56f
```

Comment (kalaiyarasiganeshalingam) [reply gradle.properties]: We can't change only the transaction's lang version. Need to update all the dependencies' lang version. Therefore, the build failed with NPE.

**Comment by kalaiyarasiganeshalingam** [comment]
We can't change only the transaction's lang version. Need to update all the dependencies' lang version.

File: transaction-ballerina/commons.bal [hunk 2]
```diff
@@ -59,7 +59,7 @@ function cleanupTransactions() returns error? {
         //foreach var twopcTxn in participatedTransactions {
             final string participatedTxnId = getParticipatedTransactionId(twopcTxn.transactionId,
                 twopcTxn.transactionBlockId);
-            if (time:utcDiffSeconds(time:utcNow(), twopcTxn.createdTime) >= 120) {
+            if (time:utcDiffSeconds(time:utcNow(), twopcTxn.createdTime) >= <decimal>120) {
```
Comment (kalaiyarasiganeshalingam) [inline transaction-ballerina/commons.bal:62]:
```suggestion
            if (time:utcDiffSeconds(time:utcNow(), twopcTxn.createdTime) >= 120d) {
```

File: gradle.properties [hunk 3]
```diff
@@ -2,7 +2,7 @@ org.gradle.caching=true
 puppycrawlCheckstyleVersion=8.18
 group=org.ballerinalang
 version=1.0.10-SNAPSHOT
-ballerinaLangVersion=2.0.0-alpha7-SNAPSHOT
+ballerinaLangVersion=2.0.0-alpha7-20210327-165900-8a5ba56f
```
Comment (grainier) [inline gradle.properties:5]:
```suggestion
ballerinaLangVersion=2.0.0-alpha7-20210327-192500-4ac1e422
```

File: gradle.properties [hunk 4]
```diff
@@ -2,7 +2,7 @@ org.gradle.caching=true
 puppycrawlCheckstyleVersion=8.18
 group=org.ballerinalang
 version=1.0.10-SNAPSHOT
-ballerinaLangVersion=2.0.0-alpha7-SNAPSHOT
+ballerinaLangVersion=2.0.0-alpha7-20210327-192500-4ac1e422
```
Comment (grainier) [inline gradle.properties:5]:
```suggestion
ballerinaLangVersion=2.0.0-alpha7-20210328-001100-42aca49c
```

**Review by lasinicl — APPROVED** [review]

**Pull request merged**